java/src/main/java/com/company/automation/fundamentals/optional_task/optional_task_01/OptionalTask01.java

```diff
@@ -0,0 +1,380 @@
+    }
+
+
+    private static int[] getIntArrayFromStringArray(String[] numbers) {
+        int[] intArray = new int[numbers.length];
+        for (int i = 0; i < intArray.length; i++) {
+            intArray[i] = Integer.parseInt(numbers[i]);
+        }
+        return intArray;
+    }
+
+
+    private static int[] getDigitsOfTheNumber(int number) {
+        int n;
+        n = String.valueOf(number).length();
+        int[] digits = new int[n];
+
+        for (int i = digits.length - 1; i >= 0; i--) {
+            int num = number / 10;
+            int digit = number - num * 10;
+            digits[i] = digit;
+            number = num;
+        }
+
+        return digits;
+    }
+
+
+    private static int getQuantityOfIntegersWithOnlyEvenDigits(String[] numbers) {
+        int quantityOfNumbersWithOnlyEvenDigits = 0;
+        int[] intArray = getIntArrayFromStringArray(numbers);
+
+        for (int val : intArray) {
+            int[] digitsOfTheNumber;
+            boolean hasOnlyEvenDigits;
+
+            digitsOfTheNumber = getDigitsOfTheNumber(val);
+            hasOnlyEvenDigits = true;
+
+            for (int item : digitsOfTheNumber) {
+                if (item % 2 != 0) {
+                    hasOnlyEvenDigits = false;
+                    break;
+                }
+            }
+
+            if (hasOnlyEvenDigits) {
+                quantityOfNumbersWithOnlyEvenDigits++;
+            }
+
+        }
+        return quantityOfNumbersWithOnlyEvenDigits;
+    }
+
+
+    private static int getQuantityOfIntegersWithEqualAmountOfEvenAndUnevenDigits(String[] numbers) {
+        int quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits = 0;
+        int[] intArray = getIntArrayFromStringArray(numbers);
+
+        for (int val : intArray) {
+            int[] digitsOfTheNumber = getDigitsOfTheNumber(val);
+            int amountOfEvenDigits = 0;
+            int amountOfUnevenDigits = 0;
+
+            for (int item : digitsOfTheNumber) {
+                if (item % 2 == 0) {
+                    amountOfEvenDigits++;
+                } else {
+                    amountOfUnevenDigits++;
+                }
+            }
+
+            if (amountOfEvenDigits == amountOfUnevenDigits) {
+                quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits++;
+            }
+        }
+
+        return quantityOfNumbersWithEqualAmountOfEvenAndUnevenDigits;
+    }
+
+
+    private static int getIntegerWithRisingDigitsSequence(String[] numbers) {
+        int integerWithRisingDigitsSequence = 0;
+        int[] intArray = getIntArrayFromStringArray(numbers);
+
+        for (int val : intArray) {
+            int[] digitsOfTheNumber;
+            boolean hasRisingDigitsSequence = false;
+            digitsOfTheNumber = getDigitsOfTheNumber(val);
+
+            if (String.valueOf(val).length() == 1) {
+                continue;
+            }
+
+            for (int i = 0; i < digitsOfTheNumber.length - 1; i++) {
```

Comment:
Try to avoid inner loops

Reply:
All loops were corrected. Inner loops were executed as additional methods.

Reply:
All loops were corrected. Inner loops were executed as additional methods.